### PR TITLE
Offchain protocl first draft

### PR DESCRIPTION
**Pull request by danprinz**
- Add off-chain structures (`payment-protocol.yaml`)
- Code generation script 
- Use the generated structures in the flow implementation
- Add options to CLI:
   - `-d` for deep link input
   - generate a new account if no private key passed, save it to file for future uses
- Wrap every response with an appropriate JSON-RPC object (protocol version, id, result, etc)

### CODE REVIEW

File: packages/server/src/routes/abort.ts [hunk 1]
```diff
@@ -1,8 +1,7 @@
 import { ResponseToolkit } from "@hapi/hapi";
-import { AbortRequest } from "@celo/payments-types";
 import { get } from "../storage";
 
-export function abort({ params }: AbortRequest, res: ResponseToolkit) {
+export function abort({ params }: any, res: ResponseToolkit) {
```
Comment (dckesler) [inline packages/server/src/routes/abort.ts:4]: A lot of the abort stuff is gone or commented out with this PR, but I assume we'll want to add it back in. I'm curious why this was switched to `any`.

File: packages/types/scripts/genereate-web-models.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
+import { EnumDeclaration, Project, SourceFile } from "ts-morph";
```
Comment (dckesler) [inline packages/types/scripts/genereate-web-models.ts:1]: I'm not sure what this file is for.

Comment (dckesler) [reply packages/types/scripts/genereate-web-models.ts:1]: Also I think it should be `generate-web-models.ts` small typo in the name

File: packages/server/src/routes/abort.ts (hunk 1 above)
Comment (danprinz) [reply packages/server/src/routes/abort.ts]: I just restored it with the new definition of the abort command, using the new interfaces as defined in the spec

File: packages/types/scripts/genereate-web-models.ts (hunk 2 above)
Comment (danprinz) [reply packages/types/scripts/genereate-web-models.ts]: It's for generating TypeScript models (interfaces) out of the open-api spec.
As this generator has some irritating "nits" about how it generates enums, I used `ts-morph` to change single-value-enum to be consist of a single member called `value`

**Pull request merged**